File: lab9-mcp-finance-go/internal/metrics/metrics.go
package metrics

import (
	"github.com/prometheus/client_golang/prometheus"
	"github.com/prometheus/client_golang/prometheus/promauto"
)

var (
	// ToolCalls счетчик вызовов инструментов
	ToolCalls = promauto.NewCounterVec(
		prometheus.CounterOpts{
			Name: "tool_calls_total",
			Help: "Общее количество вызовов инструментов",
		},
		[]string{"tool_name", "status"},
	)

	// CalculationErrors счетчик ошибок расчетов
	CalculationErrors = promauto.NewCounterVec(
		prometheus.CounterOpts{
			Name: "calculation_errors_total",
			Help: "Количество ошибок расчетов",
		},
		[]string{"tool_name", "error_type"},
	)

	// APICalls счетчик вызовов API
	APICalls = promauto.NewCounterVec(
		prometheus.CounterOpts{
			Name: "api_calls_total",
			Help: "Вызовы API инструментов",
		},
		[]string{"service", "endpoint", "status"},
	)
)

File: lab1-mcp-finance-lab/src/metrics.py
"""Метрики Prometheus для MCP сервера."""

from prometheus_client import Counter

# Публичные метрики
TOOL_CALLS = Counter(
    "tool_calls_total",
    "Общее количество вызовов инструментов",
    ["tool_name", "status"],
)

CALCULATION_ERRORS = Counter(
    "calculation_errors_total",
    "Количество ошибок расчетов",
    ["tool_name", "error_type"],
)

API_CALLS = Counter(
    "api_calls_total",
    "Вызовы API инструментов",
    ["service", "endpoint", "status"],
)
